net/fetcher.go

```diff
@@ -22,6 +30,212 @@ type Fetcher interface {
 	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, from peer.ID, done func(ts types.TipSet) (bool, error)) ([]types.TipSet, error)
 }
 
+// GraphSyncFetcher is used to fetch data over the network.  It is implemented with
+// a persistent bitswap session on a networked blockservice.
+type GraphSyncFetcher struct {
+	gs        *graphsync.GraphSync
+	validator consensus.BlockSyntaxValidator
+	bridge    ipldbridge.IPLDBridge
+	bs        bstore.Blockstore
+}
+
+// NewGraphSyncFetcher returns a BitswapFetcher wired up to the input BlockService and a newly
+// initialized persistent session of the block service.
```

Comment:
There are no block services visible here, so I don't understand. Can you clarify whether this constructor has side effects like starting goroutines or making network requests?

Reply:
outdated comment will fix.